### ISSUE TITLE
Google Sheets backend

### ISSUE THREAD

**Issue by LeaVerou** [mavoweb/plugins#75]
I would *love* a Google Sheets backend.
Not only is it useful as a primary backend, but also as a secondary export.

It would also be cool if changes from the sheet could be pushed back to the Mavo, but that would be v2 :)

**Comment by DmitrySharabin** [comment]
I was also thinking about that and have the intention to try to implement it. :)

**Comment by LeaVerou** [comment]
That would be wonderful! I hope it's possible to make Mavos of public sheets without requiring login. 

One thing to keep in mind: Do not assume that there is a 1 to 1 correspondence between a sheet and a data table. Often the same sheet contains multiple tables, pivot tables etc. It would be nice to be able to choose which one to use.

This is a very nice complex spreadsheet with over 100 sheets to explore. I'd love to make some Mavos from this data: https://docs.google.com/spreadsheets/d/1sMWXWjMujqfAREYxNbG_t1fOJKYCA6ASLwtz4pBQVTw/edit#gid=215385676

**Comment by karger** [comment]
While it may be easy to build a mavo over a google sheet, it's less 
clear how to transition an existing mavo to run over a google sheet.  
Mavo's data can be hierarchical, while google sheets are flat.  How will 
this be handled?

**Comment by DmitrySharabin** [comment]
I believe we can use the work made by others somehow. :) Since we can get data from mavo in JSON, we can then transform it into a 2D Google Sheets table with this script (or its parts): 
https://gist.github.com/paulgambill/cacd19da95a1421d3164

I haven't yet attacked the problem properly; I just made a quick search and found this gist.
I will investigate after I've finished the task I'm working on now.
